scheduler/src/cook/kubernetes/api.clj

```diff
@@ -116,30 +118,43 @@
     ; So compute the delta between the old and new and process those via the callbacks.
     ; Note as a side effect, the callbacks mutate all-pods-atom
     (doseq [task (set/union (keys namespaced-pod-name->pod) (keys old-all-pods))]
+      (log/info "In" compute-cluster-name "compute cluster, doing (startup) callback for" task)
       (doseq [callback callbacks]
         (try
-          (log/info "In" compute-cluster-name "compute cluster, doing (startup) callback for" task)
           (callback task (get old-all-pods task) (get namespaced-pod-name->pod task))
           (catch Exception e
             (log/error e "Error while processing callback for" task)))))
 
     (let [watch (WatchHelper/createPodWatch api-client (-> current-pods
                                                            .getMetadata
                                                            .getResourceVersion))]
-      (.submit kubernetes-executor ^Callable
       (fn []
         (try
           (handle-watch-updates all-pods-atom watch get-pod-namespaced-key
                                 callbacks)
           (catch Exception e
-            (log/error e "Error during watch"))
+            (log/error e "Error during pod watch for compute cluster" compute-cluster-name))
           (finally
             (.close watch)
-            (initialize-pod-watch api-client compute-cluster-name all-pods-atom cook-pod-callback))))))))
+            (initialize-pod-watch api-client compute-cluster-name all-pods-atom cook-pod-callback)))))))
 
-(defn initialize-node-watch
-  "Initialize the node watch. This fills current-nodes-atom with data and invokes the callback on pod changes."
-  [^ApiClient api-client current-nodes-atom]
+(defn initialize-pod-watch
+  "Initialize the pod watch. This fills all-pods-atom with data and invokes the callback on pod changes."
+  [^ApiClient api-client compute-cluster-name all-pods-atom cook-pod-callback]
+  ; We'll iterate trying to connect to k8s until the initilzie-pod-watch-helper returns a watch function.
+  (let [tmpfn (fn [] (try
+                       (initialize-pod-watch-helper api-client compute-cluster-name all-pods-atom cook-pod-callback)
+                       (catch Exception e
+                         (log/error e "Error during initial setup of looking at pods for" compute-cluster-name)
+                         (Thread/sleep 60000)
+                         nil)))
+        ^Callable first-success (->> tmpfn repeatedly (some identity))]
```

Comment:
Maybe `some?` instead of `identity`?

Reply:
(some some?) seems weird. Don't see anything wrong with the code as-is now.

Reply:
Agreed, there is nothing wrong with the current code, but `some?` seems more clear to me because it makes it obvious that we're looking for non-nil values. What is weird about `(some some?)`?

Reply:
Changed.

Reply:
I still see `identity`

Reply:
Pushing the changes raced with testing them. I've pushed now.

Reply:
I tested this change and now I remember why I didn't use some? Some? returns a boolean. You need identify because you need the first non-nil --- you need the value to pass to the executor.

Reply:
Ah, got it, makes sense.